From 82eb07de1b1bee0a3294c991a4fc43c650e34b88 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20=C5=A0tetiar?= <ynezz@true.cz>
Date: Mon, 6 Mar 2023 11:31:53 +0100
Subject: [PATCH] wan-manager.sh: fix not starting service
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

 root@prplOS:/# /etc/init.d/wan-manager start; ubus -t 5 wait_for X_PRPL-COM_WANManager
 Command failed: Request timed out

It seems like the init script is the culprit:

 name="wan-manager"
 name_pid="`pgrep ${name}`"

as it’s going to match the init script itself:

 root      9357  0.0  0.2   1120   524 ttyS0    S+   10:04   0:00 /bin/sh /etc/init.d/wan-manager start
 + pgrep wan-manager
 + name_pid=9357
 + '[' -z 9357 ]

and thus never start.

So lets fix it by copy&pasting defacto standard init script boilerplate
from tr181-dynamicdns service.

Fixes: PCF-855
Signed-off-by: Petr Štetiar <ynezz@true.cz>
---
 scripts/wan-manager.sh | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/scripts/wan-manager.sh b/scripts/wan-manager.sh
index 9c2d424c051b..27f2c74082ee 100644
--- a/scripts/wan-manager.sh
+++ b/scripts/wan-manager.sh
@@ -1,18 +1,18 @@
 #!/bin/sh
 [ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:0}
+# shellcheck disable=SC3045
+ulimit -c "${ULIMIT_CONFIGURATION:-0}"
 name="wan-manager"
-name_pid="`pgrep ${name}`"
 
 case $1 in
     start|boot)
-        if [ -z "${name_pid}" ]; then
-            ${name} -D
-        fi
+        ${name} -D
         ;;
     stop)
-        if [ -n "${name_pid}" ]; then
-            kill ${name_pid}
+        if [ -f /var/run/${name}.pid ]; then
+            kill "$(cat /var/run/${name}.pid)"
+        else
+            killall "${name}"
         fi
         ;;
     debuginfo)
